providers/curation/github.js

```diff
@@ -34,86 +37,71 @@ class GitHubCurationService {
     }
   }
 
-  addOrUpdate(githubUserClient, coordinates, patch) {
-    if (!patch.patch)
-      throw new Error('Cannot add or update an empty patch. Did you forget to put it in a "patch" property?')
-    const github = githubUserClient
-    const { owner, repo, branch } = this.options
+  _updateContent(coordinates, currentContent, newContent) {
+    const result = {
+      coordinates: {
+        type: coordinates.type,
+        provider: coordinates.provider,
+        namespace: coordinates.namespace === '-' ? null : coordinates.namespace,
+        name: coordinates.name
+      }
+    }
+    result.revisions = assign(get(currentContent, 'revisions') || {})
+    forIn(newContent, function (value, key) {
+      result.revisions[key] = merge(result.revisions[key] || {}, value)
+    })
+    return yaml.safeDump(result, { sortKeys: true })
+  }
+
+
+  async _writePatch(userGithub, serviceGithub, info, description, patch, branch) {
+    const { owner, repo } = this.options
+    const coordinates = new EntityCoordinates(patch.coordinates.type, patch.coordinates.provider, patch.coordinates.namespace, patch.coordinates.name)
```

Comment:
is `patch.coordinates` already an EntityCoordinates? if so, use `asRevisionless` (or whatever). If not, consider making a static helper that takes `patch.coordinates` and gives you back an EntityCoordinates that you can call `asRevisionless` on.

There is some code above where this could be useful as well.